modules/core/entity/modules/views/src/EntityViewsDataTaxonomyFilterTrait.php

```diff
@@ -22,11 +22,12 @@ protected function addReverseRelationships(array &$data, array $fields) {
       if ($field->getSettings()['target_type'] !== 'taxonomy_term') {
         continue;
       }
-      $table_name = $this->tableMapping->getFieldTableName($field->getName());
-      foreach ($data[$table_name] as $table_field_name => $field_data) {
-        $field_name = $field->getName();
-        if (isset($field_data['filter']) && strpos($table_field_name, $field_name) === 0) {
-          $data[$table_name][$table_field_name]['filter']['id'] = 'taxonomy_index_tid';
+      $field_name = $field->getName();
+      foreach ($data as $table_name => $table_data) {
+        foreach ($table_data as $table_field_name => $field_data) {
+          if (isset($field_data['filter']) && strpos($table_field_name, $field_name) === 0) {
```

Comment:
> `strpos($table_field_name, $field_name) === 0`

I'm curious why we can't check that `$table_field_name == $field_name` ? I think it could be possible that two different fields have the same prefix and this logic could inadvertently change the filter ID for the wrong field.

Consider two field names: Taxonomy reference `custom_field` and String `custom_field_raw`. I think this logic would change `custom_field_raw` to use the `taxonomy_index_tid` filter.

Reply:
but I think this approach should work! :+1:

Reply:
> Consider two field names: Taxonomy reference `custom_field` and String `custom_field_raw`. I think this logic would change `custom_field_raw` to use the `taxonomy_index_tid` filter.

That's a good point.

I'm not sure why we used `strpos($table_field_name, $field_name) === 0` originally. Here is the commit that added it, if it rings any bells: https://github.com/farmOS/farmOS/commit/7358e4d64c97f156994ddc1f821109d78d740eaf

This PR doesn't change that bit, just restructures the `foreach` so that we don't need to figure out `$table_name`.

I wonder if there was a good reason for using `strpos`... :thinking:

Reply:
> This PR doesn't change that bit, just restructures the `foreach` so that we don't need to figure out `$table_name`.

Ah but I see how that would introduce this issue, as we are no longer limiting which tables this logic can potentially run on...

Reply:
But not for all of them...

Example:

- `$table_field_name` is `units`
- `$field_name` is `units`

Reply:
Perhaps we could do:

`if (isset($field_data['filter']) && in_array($table_field_name, [$field_name, $field_name . '_target_id'])) {`

... basically checking if it's either `$field_name` or `$field_name . '_target_id'`.

Reply:
>  I think it could be possible that two different fields have the same prefix and this logic could inadvertently change the filter ID for the wrong field.

Ah ha! Sure enough I found a case where this happens in the custom codebase I'm working in, which has both `field_input` and `field_input_date` fields.